### PR TITLE
binance - fix isolated transfer

### PR DESCRIPTION
**Pull request by ttodua**
in case of `isolated`, the `params[symbol]` was still being passed (was not omitted) and overwrote the `request['symbol']` (which holds the correct value).
also, other minors change there

### CODE REVIEW

File: js/binance.js [hunk 1]
```diff
@@ -4147,21 +4147,27 @@ module.exports = class binance extends Exchange {
...
+        if (type !== undefined) {
+            request['type'] = type;
+            params = this.omit (params, 'type');
+        }
         let method = 'sapiPostAssetTransfer';
         if (request['type'] === undefined) {
+            const symbol = this.safeString (params, 'symbol');
+            if (symbol !== undefined) {
+                params = this.omit (params, 'symbol');
+            }
```
Comment (kroitor) [inline js/binance.js:4160]: In this case we must resolve the naming conflict – support both the `symbol` and the `id`, if the exchange-specific param is called `symbol` and the ccxt-param is also called `symbol`. So we must add this:

```suggestion
            let symbol = this.safeString (params, 'symbol');
            if (symbol !== undefined) {
                // support both the marketId and the unified symbol here
                symbol = this.symbol (symbol); 
                params = this.omit (params, 'symbol');
            }
```

File: js/binance.js [hunk 2]
```diff
@@ -4147,21 +4147,27 @@ module.exports = class binance extends Exchange {
             'asset': currency['id'],
             'amount': this.currencyToPrecision (code, amount),
         };
-        request['type'] = this.safeString (params, 'type');
```
Comment (kroitor) [inline js/binance.js:4150]: We cannot simply drop this line, cause `if (request['type'] === undefined) {` will throw a KeyError exception in Python after transpilation: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods

Comment (ttodua) [reply js/binance.js:4150]: uh oh, i've realized that atm.

File: js/binance.js [hunk 3]
```diff
@@ -4147,21 +4147,29 @@ module.exports = class binance extends Exchange {
...
+        const type = this.safeString (params, 'type');
+        if (type !== undefined) {
+            request['type'] = type;
+            params = this.omit (params, 'type');
+        }
```
Comment (kroitor) [inline js/binance.js:4154]: Why do we need to add these type-related lines at all? I mean, the extend-call does exactly the same thing.

```suggestion
```

**Review by kroitor — APPROVED** [review]

**Pull request merged**